### PR TITLE
Implement Contribute Page

### LINKED ISSUE #88: Migrate Contribute page to the new frontend
As per https://github.com/OWASP/Nest/issues/86 but focusing on `/projects/contribute` page only

### PR DESCRIPTION
**Pull request by harsh3dev**
Resolves #88 

Uploading Recording 2024-12-22 184922.mp4…

- [x] Add Contribute page
- [x] Add tests for contribute page
- [x] Add Search and Pagination
- [x] Consistent with new code of displaying pages

### CODE REVIEW

File: frontend/src/components/Card.tsx [hunk 1]
```diff
@@ -99,7 +99,7 @@ const Card = ({
               ))}
           </div>
           {projectName && (
-            <a href={projectLink} target="_blank" rel="noopener noreferrer">
+            <a href={projectLink} target="_blank" rel="noopener noreferrer" className="font-bold">
```
Comment (kasya) [inline frontend/src/components/Card.tsx]: @harsh3dev Longer titles don't work with this layout right now: 
Screenshot 2024-12-22 at 9 30 56 AM
Could you update it and make sure that other pages still look good?

File: frontend/src/components/Modal/Modal.tsx
Comment (kasya) [inline frontend/src/components/Modal/Modal.tsx:1]: This new modal doesn't look like the one we had before. Can we make it as close as possible to what we had before for both dark and light theme?
Current modal view in dark theme:

Screenshot 2024-12-22 at 9 33 22 AM

Prev. modal view. You can reference it on our old backend port 8000:
Screenshot 2024-12-22 at 9 33 53 AM

File: frontend/src/components/Card.tsx
Comment (kasya) [inline frontend/src/components/Card.tsx:1]: Also I noticed that the title on the card is now underneath the summary text. Could you update it to look like it was before - to have the project name underneath the title for the card?
Screenshot 2024-12-22 at 9 38 03 AM

File: frontend/src/components/Modal/Modal.tsx
Comment (kasya) [inline frontend/src/components/Modal/Modal.tsx:1]: The modal should be closed on clicking out of it. Let's update that too!
Also, I'm not sure I like the scrollable view in the modal. Can we just make it fit the full size of the available/needed screen if there is a lot of text. But if it doesn't fit the screen size - then we'll scroll?

**Review by kasya — CHANGES_REQUESTED** [review]

File: frontend/src/components/Modal/Modal.tsx
Comment (harsh3dev) [reply frontend/src/components/Modal/Modal.tsx:1]: Sure I'll implement that.

File: frontend/src/components/Card.tsx
Comment (harsh3dev) [reply frontend/src/components/Card.tsx:1]: Okay, that'll be updated.

File: frontend/src/components/Modal/Modal.tsx
Comment (harsh3dev) [reply frontend/src/components/Modal/Modal.tsx:1]: Sure Ill update the modal

File: frontend/src/components/Card.tsx (hunk 1 above)
Comment (harsh3dev) [reply frontend/src/components/Card.tsx]: I will update this after updating the existing issues.
Thanks for reviewing.

**Comment by harsh3dev** [comment]
@kasya @arkid15r 
I have implemented the suggested changes and fixed the tests that were failing.

**Comment by arkid15r** [comment]
> @kasya @arkid15r I have implemented the suggested changes and fixed the tests that were failing.

It seems the Markdown part is still missing. Let's work towards merging this one and improve it gradually.

File: frontend/src/components/Card.tsx [hunk 2]
```diff
@@ -98,11 +103,7 @@ const Card = ({
                 />
               ))}
           </div>
-          {projectName && (
-            <a href={projectLink} target="_blank" rel="noopener noreferrer">
-              {projectName}
-            </a>
-          )}
+
           {(languages || (topics && topics.length > 0) || (social && social.length > 0)) && (
```
Comment (kasya) [inline frontend/src/components/Card.tsx:112]: For issues card we also want to make sure we show topics. I believe they are already added here in the card itself, but I don't see any right now. Could you please check why is that? The topics should go under the languages labels. Screenshot of the same card at port 8000 and this PR:
Screenshot 2024-12-23 at 8 27 56 AM
Screenshot 2024-12-23 at 8 28 00 AM

File: frontend/src/components/Modal/Modal.tsx
Comment (kasya) [inline frontend/src/components/Modal/Modal.tsx:1]: This is better! I can see the markdown is still not working, but we can address that in a separate task! 👍🏼 

Another I thing I noticed with this modal is that the title and the text inside it are not exactly as they were before.
Can we make the title of the modal bolder, while reducing font-weight on the items inside the modal? Right now the `How to tackle it` thing has more weight then the title, which is not good. 
Also, can we make subtitle text muted?

Also some screenshots of current PR and modal at port 8000:
Screenshot 2024-12-23 at 8 34 33 AM
Screenshot 2024-12-23 at 8 34 39 AM

Comment (kasya) [inline frontend/src/components/Modal/Modal.tsx:1]: Also when I'm clicking out of the Modal it does close, but also the webpage is kinda of jumping a little bit. Can we fix this?

https://github.com/user-attachments/assets/ce6c788d-b9c9-4780-baad-3cf0278b0224

Comment (kasya) [inline frontend/src/components/Modal/Modal.tsx:1]: Also, can we increase the space between the title and the close button in the modal header?
Screenshot 2024-12-23 at 8 47 39 AM

**Review by kasya — CHANGES_REQUESTED** [review]
Nice updates! 👍🏼  A few comments:

**Review by kasya — CHANGES_REQUESTED** [review]
Not related to this task probably, but I just noticed that the card doesn't have a white background anymore 😮 Can we update this? 
This is how they used to look like:
Screenshot 2024-12-23 at 8 58 27 AM

**Comment by harsh3dev** [comment]
@kasya updated the components based on the suggestions.
- show topics in issues 
- make modal title bold 
- make h2 inside modal normal font-weight 
- make subtitle text muted 
- increase the space between title and close button on the top
- make the card background white
- fix the stuttering of open/close modal
Also, I am planning to implement the Markdown in the following PR if that's okay. As I am still figuring out some errors on that.

**Review by arkid15r — CHANGES_REQUESTED** [review]
@harsh3dev it's a bunch of great improvements, thank you!
Today we've merged https://github.com/OWASP/Nest/pull/247 please adjust your code for the new approach we use for Algolia requests.

Thank you!

File: frontend/src/index.css [hunk 3]
```diff
@@ -65,6 +65,52 @@ a {
   color: #1d7bd7;
 }
 
-.navlink {
-  color: inherit;
+
```
Comment (kasya) [inline frontend/src/index.css]: Why have you removed the `navlink` class from there? 🤔 
It was responsible for links color in the Header component. They are now blue and not the dark grey color.
Screenshot 2024-12-23 at 3 01 05 PM

File: frontend/src/components/Card.tsx [hunk 4]
```diff
@@ -73,6 +73,11 @@ const Card = ({
             )}
         </div>
       </div>
+      {projectName && (
+        <a href={projectLink} target="_blank" rel="noopener noreferrer" className="font-bold">
```
Comment (kasya) [inline frontend/src/components/Card.tsx]: With longer title I noticed that the meta data block gets separated from the edge of the card. If you look closely on this card the left border for the number of comments and date updated block is not touching the top of the card. It should not depend on the size of the title and always "connect" with the card.
Let me know if you need help with updating that too!
Screenshot 2024-12-23 at 3 03 18 PM

**Review by kasya — CHANGES_REQUESTED** [review]
Thanks for addressing all those comments! Noticed a couple more things after these recent updates:

**Comment by harsh3dev** [comment]
@arkid15r I have implemented all the requested changes, I have also added the new way of fetching data.
Although the tests after the changes were a bit of an obstacle but nevertheless I fixed them too.

https://github.com/user-attachments/assets/be759b45-70a9-498b-a000-a82c6a05d609

@kasya I also implemented the UI changes.
Please merge this one if it feels complete, I will handle the markdown and the minor UI changes if any in the upcoming PRs.

File: frontend/package.json
Comment (arkid15r) [inline frontend/package.json:1]: Could you double check the added dependencies -- do we really need them?

File: frontend/src/pages/Contribute.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+  return (
+    <SearchPageLayout
+      isLoaded={isLoaded}
+      totalPages={totalPages}
+      currentPage={currentPage}
+      searchQuery={searchQuery}
+      onSearch={handleSearch}
+      onPageChange={handlePageChange}
+      searchPlaceholder="Search for OWASP Issues..."
```
Comment (arkid15r) [inline frontend/src/pages/Contribute.tsx]:
```suggestion
      searchPlaceholder="Search for OWASP issues..."
```

**Review by kasya — APPROVED** [review]
@harsh3dev I pushed a couple of changes to address card styling issues that I noticed!

Nice work 👏🏼 Approved!

**Review by arkid15r — APPROVED** [review]
LGTM 👍

**Pull request merged**